src/renderer/entities/governance/lib/governanceService.ts

```diff
@@ -18,18 +21,66 @@ export const governanceService = {
   mapReferendum,
 };
 
-function mapReferendum(referendumId: string, info: ReferendaReferendumInfoConvictionVotingTally): Referendum {
+async function parseProposal(proposal: FrameSupportPreimagesBounded, api: ApiPromise): Promise<Proposal | null> {
+  let proposalHex: HexString | null = null;
+
+  if (proposal.type === 'Inline') {
+    proposalHex = proposal.data;
+  }
+
+  if (proposal.type === 'Lookup') {
+    const preimage = await api.query.preimage.preimageFor([proposal.data.hash_, proposal.data.len]);
+    if (preimage.isSome) {
+      proposalHex = u8aToHex(preimage.value);
+    }
+  }
```

Comment:
This is rather inefficient when we need to fetch this info for multiple referenda in the same list. More optimized approach is to collect all pre-images hashes we need to fetch and then fetch them in a single query
Before doing that though I'd analyze how many `Lookup` entries are there on Polkadot - if there is none (meaning uses Lookup when creating a ref) we can probably leave it as it is since this branch wont even be executed